orm/src/test/java/org/hibernate/search/test/query/dsl/DSLTest.java

```diff
@@ -351,6 +354,22 @@ public void testBoolean() throws Exception {
 		assertEquals( 1, results.size() );
 		assertEquals( "January", results.get( 0 ).getName() );
 
+		//must + disable scoring
+		query = monthQb
+				.bool()
+				.must( monthQb.keyword().onField( "mythology" ).matching( "colder" ).createQuery() )
+				.disableScoring()
+				.should( monthQb.keyword().onField( "mythology" ).matching( "colder" ).createQuery() )
```

Comment:
Why is it you have this `should()` clause here? Is it related to that optimization for single-clause boolean queries? If so, we either should document the requirement for such pseudo-phrase to enable the FILTER mode or (my preference) disable the optimization for single FILTER clauses.